src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs

```diff
@@ -51,7 +51,9 @@ public ModelSerializationExtensionsProvider() : base(Configuration.HelperNamespa
                     type: typeof(BinaryData),
                     name: _sentinelBinaryDataName)
                 {
-                    InitializationValue = BinaryDataExpression.FromObjectAsJson(Literal("__EMPTY__"))
+                    InitializationValue = BinaryDataExpression.FromBytes(
+                        new InvokeInstanceMethodExpression(
+                            LiteralU8("\"__EMPTY__\""), "ToArray", [], null, false))
```

Comment:
would it make sense to add some comments here to describe what's happening here.

Reply:
So our generator expressions currently do not have the ability of adding comments.
We could add the comments in the generator.
If we want to add comments in generated code, we might have troubles.

Reply:
I've added a quick comment explaining why we are using FromBytes and hard-coding the bytes directly.